include/bbp/reports/records.hpp

```diff
@@ -0,0 +1,70 @@
+#pragma once
+
+#include <reports/library/reportinglib.hpp>
+
+/**
+ * Provides a bridge between c-based programs in order to have access to the c++-based objects
+ */
+#ifndef __cplusplus
```

Comment:
we don't need C support, right?

Reply:
C exposed API is required because it will be called from coreneuron

Reply:
What is the plan for how libsonata will be used by coreneuron?  Why is it constrained to using only C?  I'm asking because it might make more sense to the C compatibility headers to live on the coreneuron side.

If this is a C compatibility header, why is the extension `.hpp`?